test/app-luatest/gh_7746_fiber_channel_close_test.lua

```diff
@@ -0,0 +1,62 @@
+local compat = require('tarantool').compat
```

Comment:
What just comes into my mind: shouldn't we run all test cases that lean on `compat` in separate tarantool instances (using `server:exec()`)? A change in a global state shouldn't leak into other tests.

Reply:
Created another [issue](https://github.com/tarantool/tarantool/issues/8033) for this.